tests/test_cli.py

```diff
@@ -678,6 +678,37 @@ def test_run_non_universal_newline(self):
         self.assertEqual(
             (ctx.returncode, ctx.stdout, ctx.stderr), (1, expected_out, ''))
 
+    def test_run_list_files(self):
+        self.maxDiff = None
+
+        with RunContext(self) as ctx:
+            cli.run(('--list-files', self.wd))
+        self.assertEqual(ctx.returncode, 0)
+        self.assertEqual(ctx.stdout, (
+            '{dir}/c.yaml\n'
+            '{dir}/empty.yml\n'
+            '{dir}/dos.yml\n'
+            '{dir}/warn.yaml\n'
+            '{dir}/en.yaml\n'
+            '{dir}/a.yaml\n'
+            '{dir}/s/s/s/s/s/s/s/s/s/s/s/s/s/s/s/file.yaml\n'
+            '{dir}/sub/ok.yaml\n'
+            '{dir}/sub/directory.yaml/empty.yml\n').format(dir=self.wd))
```

Comment:
Does this test pass for you? Looking at the result of GitHub Actions, it looks like there's a sorting problem.

Reply:
Mh that's strange and the tests did pass locally and the still do with Python 3.9 (Debian :roll_eyes:).

I've tried to replicate this problem by creating the same directory structure as we do in tests and voila - similar sorting problem:

```
$ yamllint --list-files temp
temp/dos.yml
temp/a.yaml
temp/warn.yaml
temp/c.yaml
temp/en.yaml
temp/empty.yml
temp/sub/ok.yaml
temp/sub/directory.yaml/empty.yml
temp/s/s/s/s/s/s/s/s/s/s/s/s/s/s/s/file.yaml
```

I also saw that related tests for `find_files_recursively` use `sorted` to sidestep the problem. I going to sort the output as well then. WDYT?

Reply:
The problem is not related to `--list-files`:

```
$ yamllint temp
temp/dos.yml
  1:1       warning  missing document start "---"  (document-start)

temp/a.yaml
  1:1       warning  missing document start "---"  (document-start)

temp/warn.yaml
  1:1       warning  missing document start "---"  (document-start)

temp/c.yaml
  1:1       warning  missing document start "---"  (document-start)

temp/en.yaml
  1:1       warning  missing document start "---"  (document-start)

temp/empty.yml
  1:1       warning  missing document start "---"  (document-start)

temp/sub/ok.yaml
  1:1       warning  missing document start "---"  (document-start)

temp/sub/directory.yaml/empty.yml
  1:1       warning  missing document start "---"  (document-start)

temp/s/s/s/s/s/s/s/s/s/s/s/s/s/s/s/file.yaml
  1:1       warning  missing document start "---"  (document-start)
```

It's a Python/Linux FS thing I believe

Reply:
I remember Ruby having a similar issue and they started to sort paths :sweat_smile: https://bugs.ruby-lang.org/issues/8709